master/internal/template/api.go

```diff
@@ -44,7 +45,10 @@ func (m *manager) put(c echo.Context) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
-	if err := yaml.Unmarshal(body, make(map[interface{}]interface{})); err != nil {
+
+	// Validate template.
+	var t expconf.ExperimentConfig
+	if err := yaml.UnmarshalStrict(body, &t, yaml.DisallowUnknownFields); err != nil {
```

Comment:
This isn't actually OK since templates can be used for command config too. 

Busy with other work now but will come back to this eventually to figure a workaround